finagle-redis/src/it/scala/com/twitter/finagle/redis/RedisIntegrationTest.scala

```diff
@@ -0,0 +1,89 @@
+package com.twitter.finagle.redis.integration
+
+import com.twitter.finagle.builder.{ClientBuilder, ServerBuilder}
+import com.twitter.finagle.redis.naggati.test.TestCodec
+import com.twitter.finagle.redis.naggati.RedisTest
+import com.twitter.finagle.redis.protocol._
+import com.twitter.finagle.redis.util._
+import com.twitter.finagle.redis.{Redis, TransactionalClient}
+import com.twitter.util.{Await, Future, Time}
+import org.jboss.netty.buffer.ChannelBuffer
+import org.scalatest.{BeforeAndAfterAll, FunSuite}
+import java.net.InetSocketAddress
+import com.twitter.finagle.Service
+import com.twitter.finagle.redis.Client
+
+trait RedisClientServerIntegrationTest extends RedisTest with BeforeAndAfterAll {
+
+  private[this] lazy val svcClient = ClientBuilder()
+    .name("redis-client")
+    .codec(Redis())
+    .hosts(RedisCluster.hostAddresses())
+    .hostConnectionLimit(2)
+    .retries(2)
+    .build()
+
+  private[this] val service = new Service[Command, Reply] {
+    def apply(cmd: Command): Future[Reply] = {
+      svcClient(cmd)
+    }
+  }
+
+  private[this] val server = ServerBuilder()
+    .name("redis-server")
+    .codec(Redis())
+    .bindTo(new InetSocketAddress(0))
+    .build(service)
+
+  override def beforeAll(): Unit = RedisCluster.start()
+  override def afterAll(): Unit = RedisCluster.stop()
+
+  protected val OKStatusReply = StatusReply("OK")
+
+  protected def withRedisClient(testCode: Service[Command, Reply] => Any) {
+    val client = ClientBuilder()
+          .name("redis-client")
+          .codec(Redis())
+          .hosts(server.boundAddress)
+          .hostConnectionLimit(1)
+          .retries(2)
+          .build()
+    Await.result(client(FlushAll))
+    try {
+      testCode(client)
+    }
+    finally {
+      client.close()
+    }
+  }
+
+  protected def assertMBulkReply(reply: Future[Reply], expects: List[String],
+    contains: Boolean = false) = Await.result(reply) match {
```

Comment:
can you break this method up w/ one param per line as in:

def lotsOfArgs(
  number: Option[String],
  typeOfPhone: String
  hoursOfAvailability: Range = 9.to(12)
)